### PR TITLE
feat: form blocks

### LINKED ISSUE #33: Form Support
Current forms on the original site:

- Contact Form (https://en-gb.mammotome.com/contact/)
- Submission Form (https://www.mammotome.com/contact/innovate-with-mammotome/)
- Study Request Form including File Upload (https://www.mammotome.com/contact/investigator-initiated-study-request/)
- Schedule a Demo Form (https://www.mammotome.com/schedule-a-demo/)
- Tech Support Ticket Form including file upload (https://www.mammotome.com/submit-a-tech-support-ticket/)

Franklin forms documentation: https://www.hlx.live/developer/forms

Some forms use file upload. How to handle this is as of yet undecided. Research as part of this ticket.

### PR DESCRIPTION
**Pull request by jalagari**
Please always provide the [GitHub issue(s)](../issues) your PR is for, as well as test URLs where your change can be observed (before and after):

Fix #33 

Test URLs:
[Contact Us](https://33-forms--mammotome--hlxsites.hlx.page/en/contact/) 
[Innovate Form](https://33-forms--mammotome--hlxsites.hlx.page/en/contact/innovate-with-mammotome/)
[Study Request](https://33-forms--mammotome--hlxsites.hlx.page/en/contact/investigator-initiated-study-request/)
[Schedule Demo](https://33-forms--mammotome--hlxsites.hlx.page/en/schedule-a-demo/) 
[Tech Support Ticket](https://33-forms--mammotome--hlxsites.hlx.page/en/submit-a-tech-support-ticket/)

### CODE REVIEW

**Comment by solaris007** [comment]
@jalagari please mention me with @solaris007 - @dominique-pfister is someone else :)

File: blocks/form/form.css [hunk 1]
```diff
@@ -0,0 +1,450 @@
...
+    --form-fieldset-border: 0;
+    --form-fieldset-marign: 0;
+    --form-fieldset-legend-color: var(--form-label-color);
+    --form-fieldset-legend-font-size: var(--form-label-font-size);
+    --form-fieldset-legend-font-weight: var(--form-label-font-weight);
+    --form-fieldset-legend-border: none;
+    --form-fieldset-legend-padding: 0;
+    --form-button-color: #FFF;
+    --form-button-font-size: var(--form-font-size-s);
+    --form-button-background-color: #5f8dda;
```
Comment (solaris007) [inline blocks/form/form.css:32]: reference to `--mt-global-color-accent-primary`

**Comment by dominique-pfister** [comment]
> @jalagari please mention me with @solaris007 - @dominique-pfister is someone else :)

which additionally doesn't have the expertise to review this PR 😉

File: blocks/form/form.css [hunk 2]
```diff
@@ -0,0 +1,450 @@
...
+    --form-label-font-size: var(--form-font-size-xs);
+    --form-label-font-weight: bold;
+    --form-fieldset-border: 0;
+    --form-fieldset-marign: 0;
+    --form-fieldset-legend-color: var(--form-label-color);
+    --form-fieldset-legend-font-size: var(--form-label-font-size);
+    --form-fieldset-legend-font-weight: var(--form-label-font-weight);
+    --form-fieldset-legend-border: none;
+    --form-fieldset-legend-padding: 0;
+    --form-button-color: #FFF;
```
Comment (solaris007) [inline blocks/form/form.css:30]: ref to `--mt-background-color-primary` or similar

File: blocks/form/form.css [hunk 3]
```diff
@@ -0,0 +1,450 @@
...
+    --form-padding: 3%;
+    --form-columns: 1;
+    --form-field-gap: 20px 0 0 0;
+    --form-invalid-border-color: #ff5f3f;
+    --form-input-padding: 0.75rem 0.6rem;
+    --form-input-font-size: 1rem;
+    --form-input-disable-color: #ccc;
+    --form-input-border-size: 1px;
+    --form-input-border-color: #818a91;
+    --form-input-background-color: #FFF;
```
Comment (solaris007) [inline blocks/form/form.css:15]: ref to --mt-background-color-primary or similar

File: blocks/form/form.css [hunk 4]
```diff
@@ -0,0 +1,450 @@
+:root {
+    --form-font-size-m: 22px;
+    --form-font-size-s: 18px;
+    --form-font-size-xs: 16px;
+    --form-background-color: white;
+    --form-padding: 3%;
+    --form-columns: 1;
+    --form-field-gap: 20px 0 0 0;
+    --form-invalid-border-color: #ff5f3f;
```
Comment (solaris007) [inline blocks/form/form.css:9]: please add (and other colors) to global colors list in `styles.css` and reference from there

File: blocks/form/form.css [hunk 5]
```diff
@@ -0,0 +1,450 @@
+:root {
+    --form-font-size-m: 22px;
+    --form-font-size-s: 18px;
+    --form-font-size-xs: 16px;
+    --form-background-color: white;
```
Comment (solaris007) [inline blocks/form/form.css:5]: ref to --mt-background-color-primary or similar

File: blocks/form/form.css [hunk 6]
```diff
@@ -0,0 +1,450 @@
...
+    --form-fieldset-legend-font-weight: var(--form-label-font-weight);
+    --form-fieldset-legend-border: none;
+    --form-fieldset-legend-padding: 0;
+    --form-button-color: #FFF;
+    --form-button-font-size: var(--form-font-size-s);
+    --form-button-background-color: #5f8dda;
+    --form-button-background-hover-color: #035fe6;
+    --form-button-border: 2px solid transparent;
+    --form-button-padding:15px 50px;
+    --form-upload-color: #FFF;
```
Comment (solaris007) [inline blocks/form/form.css:36]: ref to --mt-background-color-primary or similar

File: blocks/form/form.css [hunk 7]
```diff
@@ -0,0 +1,450 @@
...
+    text-align: center;
+}
+
+main .form button {
+    color: var(--form-button-color);
+    background: var(--form-button-background-color);
+    border: var(--form-button-border);
+    padding: var(--form-button-padding);
+    font-size: var(--form-button-font-size);
+    font-weight: bold;
```
Comment (solaris007) [inline blocks/form/form.css:154]: please reference `--mt-font-weight-bold` or simialr

File: blocks/form/form.css [hunk 8]
```diff
@@ -0,0 +1,450 @@
...
+    font-style: var(--form-paragraph-font-style);
+    color: var(--form-paragraph-color);
+    margin: var(--form-paragraph-margin);
+}
+
+.technical-inquiry  {
+    --form-fieldset-legend-font-size: 1.75rem;
+    --form-fieldset-legend-font-weight: 400;
+    --form-fieldset-legend-border: 1px solid rgba(0, 0, 0, 10%);
+    --form-fieldset-legend-color: #84329B;
```
Comment (solaris007) [inline blocks/form/form.css:187]: --mt-global-color-base-primary

File: blocks/form/form.css [hunk 9]
```diff
@@ -0,0 +1,450 @@
...
+    --form-fieldset-legend-padding: 0px 0px 20px 0px
+}
+
+.innovate,
+.study-request  {
+    --form-fieldset-legend-font-size: 1.50rem;
+    --form-fieldset-legend-font-weight: 500;
+    --form-fieldset-legend-border: 2px dashed #fff;
+    --form-background-color: #84329b;
```
Comment (solaris007) [inline blocks/form/form.css:196]: --mt-global-color-base-primary

File: blocks/form/form.css [hunk 10]
```diff
@@ -0,0 +1,450 @@
...
+    --form-fieldset-legend-border: 1px solid rgba(0, 0, 0, 10%);
+    --form-fieldset-legend-color: #84329B;
+    --form-fieldset-legend-padding: 0px 0px 20px 0px
+}
+
+.innovate,
+.study-request  {
+    --form-fieldset-legend-font-size: 1.50rem;
+    --form-fieldset-legend-font-weight: 500;
+    --form-fieldset-legend-border: 2px dashed #fff;
```
Comment (solaris007) [inline blocks/form/form.css:195]: --mt-background-color-primary

File: blocks/form/form.css [hunk 11]
```diff
@@ -0,0 +1,450 @@
...
+    --form-background-color: #84329b;
+    --form-label-color: #fff;
```
Comment (solaris007) [inline blocks/form/form.css:197]: --mt-background-color-primary

File: blocks/form/form.css [hunk 12]
```diff
@@ -0,0 +1,450 @@
...
+    --form-paragraph-color: #fff;
+    --form-fieldset-legend-color: #fff;
```
Comment (solaris007) [inline blocks/form/form.css:199]: --mt-background-color-primary

File: blocks/form/form.css [hunk 13]
```diff
@@ -0,0 +1,450 @@
...
+main .study-request form fieldset fieldset {
+    --form-fieldset-legend-color: var(--form-label-color);
+    --form-fieldset-legend-font-size: var(--form-label-font-size);
+    --form-fieldset-legend-font-weight: var(--form-label-font-weight);
+    --form-fieldset-legend-border: none;
+    --form-fieldset-legend-padding: 0;
+}
+
+.form-PIandContactTitle {
+    font-weight: bold;
```
Comment (solaris007) [inline blocks/form/form.css:223]: --mt-font-weight-bold

File: blocks/form/form.css [hunk 14]
```diff
@@ -0,0 +1,450 @@
...
+
+.form-message,
+.form-consentMsg {
+    font-size:  var(--mt-font-size-90);
+    border-bottom: solid 2px #eee;
+}
+
+.form-message {
+    font-size:  var(--mt-font-size-130);
+    font-weight: bold;
```
Comment (solaris007) [inline blocks/form/form.css:274]: --mt-font-weight-bold

File: styles/styles.css [hunk 15]
```diff
@@ -473,4 +473,36 @@ main .section.our-history .year.highlight .icon-calendar {
     height: 85%;
   }
 
+main:has(.schedule-demo, .contact) {
+  padding: 05% 0% 10% 0%;
+  background-color: #84329b;
+  background-image: url(https://www.mammotome.com/wp-content/uploads/2019/11/Mammotome-BG_Pattern-1.svg);
```
[the remark this answers is not in the document]
Comment (solaris007) [reply styles/styles.css:479]: this may apply to all styles in this patch for styles.css

[the remark this answers is not in the document]
Comment (jalagari) [reply styles/styles.css:479]: Removed background image. 
For now keeping background-color just to aware we need to update it

File: blocks/form/form.css [hunk 16]
```diff
@@ -0,0 +1,458 @@
...
+    --form-label-font-size: var(--form-font-size-xs);
+    --form-label-font-weight: bold;
+    --form-fieldset-border: 0;
+    --form-fieldset-marign: 0;
+    --form-fieldset-legend-color: var(--form-label-color);
+    --form-fieldset-legend-font-size: var(--form-label-font-size);
+    --form-fieldset-legend-font-weight: var(--form-label-font-weight);
+    --form-fieldset-legend-border: none;
+    --form-fieldset-legend-padding: 0;
+    --form-button-color: var(--mt-background-color-primary);;
```
Comment (solaris007) [inline blocks/form/form.css:30]: double semi-colon

File: blocks/form/form.css [hunk 17]
```diff
@@ -0,0 +1,458 @@
...
+    --form-background-color: var(--mt-background-color-primary);
+    --form-padding: 3%;
+    --form-columns: 1;
+    --form-field-gap: 20px 0 0 0;
+    --form-invalid-border-color: var(--mt-global-color-orange);
+    --form-input-padding: 0.75rem 0.6rem;
+    --form-input-font-size: 1rem;
+    --form-input-disable-color: #ccc;
+    --form-input-border-size: 1px;
+    --form-input-border-color: #818a91;
```
Comment (solaris007) [inline blocks/form/form.css:14]: please add or re-use var for color

File: blocks/form/form.css [hunk 18]
```diff
@@ -0,0 +1,458 @@
...
+    --form-input-background-color: var(--mt-background-color-primary);
+    --form-paragraph-color: #666;
```
Comment (solaris007) [inline blocks/form/form.css:16]: please add or re-use var for color

File: blocks/form/form.css [hunk 19]
```diff
@@ -0,0 +1,458 @@
...
+    --form-paragraph-margin: 0 0 0.9rem;
+    --form-paragraph-font-style: none;
+    --form-paragraph-font-size: var(--form-font-size-xs);
+    --form-label-color: #666;
```
Comment (solaris007) [inline blocks/form/form.css:20]: please add or re-use var for color

File: blocks/form/form.css [hunk 20]
```diff
@@ -0,0 +1,458 @@
...
+    width: var(--form-submit-width);
+}
+
+main .form :is(input:disabled, textarea:disabled, select:disabled, button:disabled) {
+    background-color: var(--form-input-disable-color);
+}
+
+.technical-inquiry  {
+    --form-fieldset-legend-font-size: 1.75rem;
+    --form-fieldset-legend-font-weight: 400;
```
Comment (solaris007) [inline blocks/form/form.css:193]: please reference this var to the relevant font weight var in styles.css

File: blocks/form/form.css [hunk 21]
```diff
@@ -0,0 +1,458 @@
...
+    --form-fieldset-legend-border: 1px solid rgb(0 0 0 / 10%);
+    --form-fieldset-legend-color: var(--mt-global-color-base-primary);
+    --form-fieldset-legend-padding: 0px 0px 20px 0px
+}
+
+.innovate,
+.study-request  {
+    --form-fieldset-legend-font-size: 1.50rem;
+    --form-fieldset-legend-font-weight: 500;
```
Comment (solaris007) [inline blocks/form/form.css:202]: please reference this var to the relevant font weight var in styles.css

File: blocks/form/form.css [hunk 22]
```diff
@@ -0,0 +1,458 @@
...
+    --form-label-color: var(--mt-background-color-primary);
+    --form-paragraph-color: var(--mt-background-color-primary);
+    --form-fieldset-legend-color: var(--mt-background-color-primary);
+    --form-fieldset-legend-padding: 1em 0;
+}
+
+.innovate {
+    --form-fieldset-legend-border: 1px solid gray;
+    --form-fieldset-legend-font-size: 2.15rem;
+    --form-fieldset-legend-font-weight: 700;
```
Comment (solaris007) [inline blocks/form/form.css:214]: please reference this var to the relevant font weight var in styles.css

File: blocks/form/form.css [hunk 23]
```diff
@@ -0,0 +1,458 @@
...
+main .technical-inquiry form {
+    display: flex;
+    flex-direction: column;
+}
+
+.contact,
+.schedule-demo {
+    --form-paragraph-color: var(--mt-global-color-black);
+    --form-label-color: var(--mt-global-color-black);
+    --form-input-background-color: #F7F7F7;
```
Comment (solaris007) [inline blocks/form/form.css:267]: please add or re-use var for color

File: blocks/form/form.css [hunk 24]
```diff
@@ -0,0 +1,458 @@
...
+    --form-input-border-size: 1px;
+    --form-input-border-color: #ECECEC;
```
Comment (solaris007) [inline blocks/form/form.css:269]: please add or re-use var for color

File: blocks/form/form.css [hunk 25]
```diff
@@ -0,0 +1,458 @@
...
+    display: block;
+    margin: 0 auto;
+    text-align: center;
+    margin-top: 15px;
+    margin-bottom: 15px;
+}
+
+main .study-request input[type='file'] {
+    background: rgb(var(--mt-global-color-black) / 2%);
+    border: 1px dashed #ccc;
```
Comment (solaris007) [inline blocks/form/form.css:249]: please add or re-use var for color

File: blocks/form/form.css [hunk 26]
```diff
@@ -0,0 +1,458 @@
...
+    --form-label-color: var(--mt-global-color-black);
+    --form-input-background-color: #F7F7F7;
+    --form-input-border-size: 1px;
+    --form-input-border-color: #ECECEC;
+}
+
+main .form form .form-message,
+.form-consent {
+    font-size:  var(--mt-font-size-90);
+    border-bottom: solid 2px #eee;
```
Comment (solaris007) [inline blocks/form/form.css:275]: please add or re-use var for color

File: blocks/form/form.css [hunk 27]
```diff
@@ -0,0 +1,458 @@
...
+   width: calc(100% - 16px);
+}
+
+@media (min-width: 576px) {
+    .form-container {
+        max-width: 540px;
+    }
+}
+  
+  @media (min-width: 768px) {
```
Comment (solaris007) [inline blocks/form/form.css:345]: indentation

File: blocks/form/form.css [hunk 28]
```diff
@@ -0,0 +1,458 @@
...
+        margin: 0 auto 10px;
+    }
+  
+     form {
+        width: unset;
+        flex-flow: wrap row;
+    }
+  }
+  
+  @media (min-width: 1200px), (min-width: 992px) {
```
Comment (solaris007) [inline blocks/form/form.css:357]: indendation

File: styles/styles.css [hunk 29]
```diff
@@ -432,6 +438,34 @@ main .section.our-history .year.highlight .icon-calendar {
   background-color: var(  --mt-our-history-color-primary);
 }
 
+main:has(.schedule-demo, .contact) {
+  padding: 05% 0% 10%;
+  background-color: #84329b;
```
Comment (solaris007) [inline styles/styles.css:443]: please add or re-use var for color

File: styles/styles.css [hunk 30]
```diff
@@ -432,6 +438,34 @@ main .section.our-history .year.highlight .icon-calendar {
...
+  background-position: top center;
+  background-repeat: repeat;
+  background-size: 60px auto;
+}
+
+main .innovate .form,
+main .study-request .form {
+  background-color: #84329b;
```
Comment (solaris007) [inline styles/styles.css:451]: please add or re-use var for color

**Review by solaris007 — COMMENTED** [review]
just minor comments

File: blocks/form/form.css [hunk 31]
```diff
@@ -0,0 +1,450 @@
...
+
+main .form .field-description {
+    color: var(--form-label-color);
+    font-size: var(--form-font-size-xs);
+}
+
+main .form :is(input, textarea, select) {
+    background-color: var(--form-input-background-color);
+    border: var(--form-input-border-size) solid var(--form-input-border-color);
+    width: calc(100% - 16px);
```
[the remark this answers is not in the document]
Comment (jalagari) [reply blocks/form/form.css:74]: I have copied it from form block looks like its for reducing the margin and padding set on input field

**Review by solaris007 — APPROVED** [review]

**Pull request merged**